test/parallel/test-tls-handshake-exception.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+// Verify that exceptions from a callback don't result in
+// failed CHECKs when trying to print the exception message.
+
+// This test is convoluted because it needs to trigger a callback
+// into JS land at just the right time when an exception is pending,
+// and does so by exploiting a weakness in the streams infrastructure.
+// I won't shed any tears if this test ever becomes invalidated.
```

Comment:
I'm sorry if I'm being dense, but can you explain what you mean by this comment? I'm not sure what you mean exactly by "becomes invalidated". Does that mean "becomes unnecessary" in this context? Or maybe it means "stops working"? Or something else?

Reply:
Unnecessary, yes, as in "no longer tests this particular condition because node's internals changed."

Reply:
Ah, literally invalidated like you wrote initially, just I was thinking about it wrong. I get it now. Thanks.